apps/condo/domains/notification/tasks.js

```diff
@@ -1,118 +1,166 @@
 
-    const processingMeta = { dv: 1, transport, step: 'init' }
     await Message.update(keystone, message.id, {
         ...baseAttrs,
         status: MESSAGE_PROCESSING_STATUS,
         deliveredAt: null,
         processingMeta,
     })
 
-    try {
-        const adapter = TRANSPORTS[transport]
-        const messageContext = await adapter.prepareMessageToSend(message)
-        processingMeta.step = 'prepared'
-        processingMeta.messageContext = messageContext
+    const failedMeta = []
+
+    for (const transport of transports) {
+
+        if (message.type !== 'REGISTER_NEW_USER') console.log('deliverMessage:', { messageId, message, transport })
 
-        const [isOk, deliveryMetadata] = await _sendMessageByAdapter(transport, adapter, messageContext)
-        processingMeta.deliveryMetadata = deliveryMetadata
-        processingMeta.step = (isOk) ? 'delivered' : 'notDelivered'
-        if (!isOk) throw Error('Transport send result is not OK. Check deliveryMetadata')
-    } catch (e) {
-        console.error(e)
-        processingMeta.error = e.stack || String(e)
+        try {
+            const adapter = TRANSPORTS[transport]
+            const messageContext = await adapter.prepareMessageToSend(message)
 
+            if (message.type !== 'REGISTER_NEW_USER') console.log('deliverMessage try:', { transport, message, messageContext })
+
+            const [isOk, deliveryMetadata] = await _sendMessageByAdapter(transport, adapter, messageContext)
+
+            if (isOk) {
+                processingMeta.messageContext = messageContext
+                processingMeta.deliveryMetadata = deliveryMetadata
+                processingMeta.step = 'delivered'
+                processingMeta.transport = transport
+                break
+            } else {
+                console.error('Transport send result is not OK. Check deliveryMetadata', deliveryMetadata)
+                failedMeta.push({
+                    error: 'Transport send result is not OK. Check deliveryMetadata',
+                    transport,
+                    messageContext,
+                    deliveryMetadata,
+                })
+            }
+        } catch (e) {
+            console.error(e)
+
+            failedMeta.push({ transport, errorStack: e.stack, error: String(e) })
+        }
+    }
+
+    // message delivered either directly or by fallback transport
+    if (processingMeta.step === 'delivered') {
+        await Message.update(keystone, message.id, {
+            ...baseAttrs,
+            status: MESSAGE_DELIVERED_STATUS,
+            deliveredAt: new Date().toISOString(),
+            processingMeta,
+        })
+    } else {
         await Message.update(keystone, message.id, {
             ...baseAttrs,
             status: MESSAGE_ERROR_STATUS,
             deliveredAt: null,
-            processingMeta,
+            processingMeta: failedMeta,
         })
 
-        throw e
```

Comment:
You need to throw the error! If the message is not delivered. This will force this task to be executed again